devcontainer.md

```diff
@@ -0,0 +1,109 @@
+For information on running demos and tests using provided shell scripts, see [DevReadMe](/DevReadMe.md) readme.
+
+# ACA-Py Development with Dev Container
+The following guide will get you up and running and developing/debugging ACA-Py as quickly as possible. 
+We provide a [`devcontainer`](https://containers.dev) and will use [`VS Code`](https://code.visualstudio.com) to illustrate.
+
+By no means is ACA-Py limited to these tools; they are merely examples.  
+
+## Further Reading and Links
+
+* Development Containers (devcontainers): [https://containers.dev](https://containers.dev)
+* Visual Studio Code: [https://code.visualstudio.com](https://code.visualstudio.com)
+* Dev Containers Extension: [marketplace.visualstudio.com](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers)
+* Docker: [https://www.docker.com](https://www.docker.com)
+* Docker Compose: [https://docs.docker.com/compose/](https://docs.docker.com/compose/)
+
+
+## Devcontainer
+
+> What are Development Containers?
+> 
+> A Development Container (or Dev Container for short) allows you to use a container as a full-featured development environment. It can be used to run an application, to separate tools, libraries, or runtimes needed for working with a codebase, and to aid in continuous integration and testing. Dev containers can be run locally or remotely, in a private or public cloud.
+
+see [https://containers.dev](https://containers.dev).
+
+In this guide, we will use [Docker](https://www.docker.com) and [Visual Studio Code](https://code.visualstudio.com) with the [Dev Containers Extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers) installed, please set your machine up with those. As of writing, we used the following: 
+
+- Docker Version: 20.10.24
+- VS Code Version: 1.79.0
+- Dev Container Extension Version: v0.295.0
+
+### Open ACA-Py in the devcontainer
+
+There are 2 ways to open ACA-Py into the devcontainer and start developing.
+
+1. Open Visual Studio Code, and use the Command Palette to open the devcontainer.
+	
+	```
+	> Dev Containers: Open Folder in Container...
+	```
+
+2. Open the root of this repository and you should be prompted to `Reopen in Container`.
+
+*NOTE* follow any prompts to install `Python Extension` or reload window for `Pylance` when first building the container.
+
+#### devcontainer.json
+
+When the [.devcontainer/devcontainer.json](.devcontainer/devcontainer.json) is opened, you will see it building... what it is building is a Python 3.9 image and loading it with ACA-Py requiments (and black). Since this is a Docker container, we will also open ports `9060` and `9061` which will allow you to run ACA-Py with those ports available to your `localhost`. We also load a few Visual Studio settings (for running Pytests and formatting with Flake and Black).
+
+In VS Code, open a Terminal, you should be able to run the following commands:
```

Comment:
Defaults a bash terminal (assume this comes from a default from the devcontainer) for me, think that's correct but not sure if bash specifically is required for the instructions? IE, should this say "open a Bash Terminal"

Reply:
yeah, it'll be bash for all. noted in the readme now.